### PR TITLE
Avoid double requests

### PR DESCRIPTION
**Pull request by adrianschroeter**

### CODE REVIEW

**Comment by bgeuken** [comment]
LGTM

File: src/api/app/models/bs_request.rb [hunk 1]
```diff
@@ -154,6 +154,9 @@ def self.new_from_xml(xml)
     if hashed['id']
       theid = hashed.delete('id') { raise 'not found' }
       theid = Integer(theid)
+      if BsRequest.find_by_number(theid)
```
Comment (hennevogel) [inline src/api/app/models/bs_request.rb]: This is a validation.

File: src/api/app/models/bs_request.rb [hunk 2]
```diff
@@ -189,7 +192,7 @@ def self.new_from_xml(xml)
 
       request.commenter = state.delete('who')
       unless request.commenter
-        raise 'no one logged in and no user in request' unless User.current
+        raise SaveError, 'no one logged in and no user in request' unless User.current
```
Comment (hennevogel) [inline src/api/app/models/bs_request.rb]: Same here. A [validation on the presence](http://guides.rubyonrails.org/active_record_validations.html#presence).

File: src/api/app/models/bs_request.rb (hunk 1 above)
Comment (adrianschroeter) [inline src/api/app/models/bs_request.rb]: sure, but it would not give a proper error message, would it?

Comment (bgeuken) [inline src/api/app/models/bs_request.rb]: The validation messages can be customized.

Comment (adrianschroeter) [inline src/api/app/models/bs_request.rb]: so, how would it look alike to get the same? (http error code and OBS xml status error with code and details)

Comment (hennevogel) [inline src/api/app/models/bs_request.rb]: You validate in the model. When you save the data in the controller you check the return and [handle it](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/controllers/webui/configuration_controller.rb#L22).

Comment (adrianschroeter) [inline src/api/app/models/bs_request.rb]: modifing user data in validation is IMHO not a good thing (in the commentor case below).

So, I keep this code but add validations in addition to be save, okay?

Comment (hennevogel) [inline src/api/app/models/bs_request.rb]: Huh? Who is modifying user data?

**Comment by adrianschroeter** [comment]
I started to move handling to validations, but I stopped after the patch becomes too large for now.

So, this is only the database protection without anything else. Better crash then broken data or invalid api answers.

**Pull request merged**